src/Filters/Has.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace LaravelJsonApi\Eloquent\Filters;
+
+use LaravelJsonApi\Eloquent\Contracts\Filter;
+use LaravelJsonApi\Eloquent\Schema;
+
+class Has implements Filter
+{
+    use Concerns\DeserializesValue;
+    use Concerns\IsSingular;
+
+    /**
+     * @var string
+     */
+    private string $relationName;
+
+    private Schema $schema;
+
+    /**
+     * SearchFilter constructor.
+     *
+     * @param string $relationName
+     * @param array $columns
+     */
+    public function __construct(string $relationName, Schema $schema)
```

Comment:
Several things here.

Can you swap the arguments so that the schema is first, JSON:API field name is second. Please rename the `$relationName` argument to `$fieldName`, just so it is clearer we're referring to the JSON:API field name and not the Eloquent relation name.

Reply:
The other thing is the filter key may not be the same as the JSON:API field name. Therefore you need to add a third argument, `$key`. That can be nullable - if it is not provided, the key should be the same as the JSON:API field name.